CMSIS/DoxyGen/Core/src/Ref_cm4_simd.txt

```diff
@@ -1433,22 +1433,50 @@ uint32_t __SXTAB16(uint32_t val1, uint32_t val2);
 
 
 /**************************************************************************************************/
-/** 
+/**
+    \brief      Rotate right, followed by sign extension of two 8-bits with add to 16-bits
+
+    \details    This function enables you to rotate the second operand by 8/16/24 bit as specified by the third
+          operand, extract two 8-bit values from the rotated result (at bit positions [7:0] and [23:16]),
+          sign-extend them to 16-bits each, and add the results to the first operand.
+
+    \param      val1    two 16-bit values in val1[15:0] and val1[31:16]
+    \param      val2    two 8-bit values in val[7:0] and val[23:16] to be sign-extended post rotation
+    \param      rotate  number of bits to rotate val2. Only 8,16 and 24 are accepted
```

Comment:
Is it? I guess you can put everything here as long it's a constant value. If its one of 8, 16 or 24 the optimized instruction `SXTAB16_RORn` is generated. For other rotate values one ends up with separate `ROR` and `SXTAB16` instructions.

Reply:
That is correct.. I'll update the PR to make that distinction and update the documentation for SXTB16_RORn as well.

Reply:
The SXTAB16 instruction only allows the rotate values 8, 16, 24.
Shall the __SXTAB16_RORn support allow more rotate values?

Reply:
I guess "Only 8,16 and 24 are accepted" is a bit misleading as __SXTAB16_RORn can accept other constant values as well, except that it won't be optimal or give any additional value. 

I can phrase it either as

"constant rotate value of 8, 16 and 24 can be optimally used with a single __SXTAB16 instruction. Any other valid constant rotate value will result in use of two instructions, __ROR and __SXTAB16"

or keep it simple and say,

"optimal and intended use is for constant rotate value of 8, 16 or 24"

Any preference?

Reply:
>constant rotate value of 8, 16 and 24 can be optimally used with a single __SXTAB16 instruction. Any other valid constant rotate value will result in use of two instructions, __ROR and __SXTAB16

Reads good to me.

Reply:
`constant rotate value of 8, 16 and 24 can be optimally used with a single __SXTAB16 instruction. Any other valid constant rotate value will result in use of two instructions, __ROR and __SXTAB16`

is perfect.